pip/pip-420.md

```diff
@@ -0,0 +1,253 @@
+            .topic(topic)
+            .create();
+
+    Consumer<User> consumer = client.newConsumer(schema)
+            .topic(topic)
+            .subscriptionName("sub")
+            .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
+            .subscribe();
+
+    for (int i = 0; i < 10; i++) {
+        producer.send(new User("name-" + i, 10 + i));
+    }
+
+    for (int i = 0; i < 10; i++) {
+        Message<User> message = consumer.receive();
+        consumer.acknowledge(message);
+    }
+
+    client.close();
+}
+```
+
+Messaging protocols changes
+
+Ignore schema info while creating Producer, ignore register schema before sending messages, the schema management 
+```java
+public class ProducerImpl<T> extends ProducerBase<T> implements TimerTask, ConnectionHandler.Connection {
+
+    @Override
+    public CompletableFuture<Void> connectionOpened(final ClientCnx cnx) {
+        // ...
+
+        if (schema != null && schema.getSchemaInfoProvider() != null && schema.getSchemaInfoProvider().isExternal()) {
+            try {
+                // set schemaInfo to null if the schema info provider is external
+                schemaInfo = null;
+            } catch (Exception e) {
+                log.error("Failed to register schema.", e);
+                throw new RuntimeException(e);
+            }
+        }
+
+        // send create producer request
+    }
+
+    private void tryRegisterSchema(ClientCnx cnx, MessageImpl msg, SendCallback callback, long expectedCnxEpoch) {
+        // ...
+
+        if (schema.getSchemaInfoProvider().isExternal()) {
+            // don't register schema if external schema registry service, the register method can be integrated in the messaging encode method.
+            return;
+        }
+        // getOrCreateSchemaAsync
+    }
+
+}
+```
+
+Ignore schema info while adding consumer subscription.
+```java
+public class ConsumerImpl<T> extends ConsumerBase<T> implements ConnectionHandler.Connection {
+
+    @Override
+    public CompletableFuture<Void> connectionOpened(final ClientCnx cnx) {
+        // ...
+
+        // only external schema info provider need to register schema
+        if (schema.getSchemaInfoProvider() != null && schema.getSchemaInfoProvider().isExternal()) {
+            try {
+                // don't set schema info for schema registry schema
+                si = null;
+            } catch (Exception e) {
+                log.error("Failed to register schema.", e);
+                throw new RuntimeException(e);
+            }
+        }
+
+    }
+
+}
+```
+
+## Public-facing Changes
+
+Add new methods for `SchemaInfoProvider` interface.
+The `SchemaInfoProvider` provides necessary params for connecting to the external schema registry service with the method `getConfigs`.
+If the schema info provider is external, the new producer command, consumer subscribe command will treat the schema as bytes schema, the broker will ignore schema validation.
```

Comment:
We should tell the broker that the producer or consumer is using an external schema registry to access this topic. Otherwise, if a Pulsar producer with built-in schema produces messages to this topic, it will succeed.